snuba/utils/streams/kafka.py

```diff
@@ -660,45 +651,25 @@ def build_kafka_consumer_configuration(
     }
 
 
-class CommitCodec(Codec[KafkaPayload, Commit]):
-    def encode(self, value: Commit) -> KafkaPayload:
-        return KafkaPayload(
-            f"{value.partition.topic.name}:{value.partition.index}:{value.group}".encode(
-                "utf-8"
-            ),
-            f"{value.offset}".encode("utf-8"),
-        )
-
-    def decode(self, value: KafkaPayload) -> Commit:
-        key = value.key
-        if not isinstance(key, bytes):
-            raise TypeError("payload key must be a bytes object")
-
-        val = value.value
-        if not isinstance(val, bytes):
-            raise TypeError("payload value must be a bytes object")
-
-        topic_name, partition_index, group = key.decode("utf-8").split(":", 3)
-        offset = int(val.decode("utf-8"))
-        return Commit(group, Partition(Topic(topic_name), int(partition_index)), offset)
+# XXX: This must be imported after `KafkaPayload` to avoid a circular import.
+from snuba.utils.streams.synchronized import Commit, commit_codec
```

Comment:
See description re: coupling, there is also a more descriptive comment on the alternative approach in `snuba/utils/streams/synchronized.py`.